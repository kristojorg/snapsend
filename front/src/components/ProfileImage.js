//@flow
import * as React from 'react';
import Avatar from 'material-ui/Avatar';
import AddAPhoto from 'material-ui-icons/AddAPhoto';
import { uploadImage } from '../uploadImage';
import { CircularProgress } from 'material-ui/Progress';

type P = {
  style?: any,
  handlePicChange: string => void,
  props?: any,
  profilePic: ?string,
};
type S = {
  url: ?string,
  loading: boolean,
};

export default class ProfilePic extends React.Component<P, S> {
  state = {
    url: null,
    loading: false,
  };

  inputRef: ?HTMLInputElement;

  openPicker = () => {
    this.inputRef && this.inputRef.click();
  };

  handleFileChange = async (event: SyntheticEvent<HTMLInputElement>) => {
    const { currentTarget } = event;
    const { files } = currentTarget;

    if (files && files[0]) {
      // set a loading indicator
      this.setState({ loading: true });
      // send to filestack
      const res = await uploadImage(files[0]);
      // then call the change handleer
      this.props.handlePicChange(res.url);
    }
  };

  render() {
    const { profilePic, style = {}, handlePicChange, ...props } = this.props;
    const { loading } = this.state;
    return (
      <React.Fragment>
        <Avatar
          style={{ width: 100, height: 100, ...style }}
          onClick={this.openPicker}
          src={profilePic}
          {...props}
        >
          {!profilePic &&
            (loading ? <CircularProgress color="secondary" /> : <AddAPhoto />)}
        </Avatar>
        <input
          type="file"
          accept="image/*"
          onChange={this.handleFileChange}
          ref={r => (this.inputRef = r)}
          style={{ display: 'none' }}
        />
      </React.Fragment>
    );
  }
}
