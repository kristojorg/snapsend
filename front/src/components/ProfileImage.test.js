import React from 'react';
import ProfileImage from './ProfileImage';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

/**
 * Button just does what button does
 * Only it does it w/ special style.
 */

describe('ProfileImage Component', () => {
  it('ProfileImage renders correctly', () => {
    const tree = renderer
      .create(<ProfileImage>Click me!</ProfileImage>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('ProfileImage renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProfileImage>Click me!</ProfileImage>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
