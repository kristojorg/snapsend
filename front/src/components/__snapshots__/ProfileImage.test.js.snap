// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage Component ProfileImage renders correctly 1`] = `
Array [
  <div
    className="MuiAvatar-root-1 MuiAvatar-colorDefault-2"
    onClick={[Function]}
    style={
        Object {
            "height": 100,
            "width": 100,
          }
    }
>
    <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-4"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
    >
        <path
            d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
        />
    </svg>
</div>,
  <input
    accept="image/*"
    onChange={[Function]}
    style={
        Object {
            "display": "none",
          }
    }
    type="file"
/>,
]
`;
